dataikuapi/dss/ml.py

```diff
@@ -829,17 +829,28 @@ def set_range(self, min=None, max=None, nb_values=None):
 
     @property
     def range(self):
-        return Range(self)
+        return RangeSettings(self)
 
 
 class Range(object):
 
+    def __init__(self, min, max, nb_values=3):
+        self.min = min
+        self.max = max
+        self.nb_values = nb_values
+
+    def __repr__(self):
+        return "Range(min={}, max={}, nb_values={})".format(self.min, self.max, self.nb_values)
+
+
+class RangeSettings(object):
```

Comment:
The main idea is that the user should not instanciate the `RangeSettings` class himself. Maybe nest this class inside `NumericalHyperparameterSettings` and call it using `self.RangeSettings` , to make it very clear that it has no purpose outside of `NumericalHyperparameterSettings`?

Reply:
Nesting sounds like the best solution to "hide" the `RangeSettings` class indeed, thanks for the suggestion !